src/main/java/com/gradle/enterprise/conventions/PublishingConfigurationAction.java

```diff
@@ -0,0 +1,32 @@
+package com.gradle.enterprise.conventions;
+
+import com.gradle.develocity.agent.gradle.scan.BuildScanPublishingConfiguration;
+import org.gradle.api.Action;
+
+public abstract class PublishingConfigurationAction implements Action<BuildScanPublishingConfiguration> {
+
+    public static final PublishingConfigurationAction PUBLISH_IF_AUTHENTICATED = new PublishingConfigurationAction("publishWhenAuthenticated") {
+        @Override
+        public void execute(BuildScanPublishingConfiguration publishing) {
+            publishing.onlyIf(BuildScanPublishingConfiguration.PublishingContext::isAuthenticated);
```

Comment:
🤔 Could we be a little bit better here now? I'd say we publish if
- We are authenticated
- or someone requests `--scan` via the command line.

WDYT?